tests/components/balboa/__init__.py

```diff
@@ -0,0 +1,167 @@
+"""Test the Balboa Spa Client integration."""
+import asyncio
+from unittest.mock import patch
+
+from homeassistant.components.balboa.const import DOMAIN as BALBOA_DOMAIN
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+BALBOA_DEFAULT_PORT = 4257
+TEST_HOST = "balboatest.localdomain"
+
+
+async def init_integration(hass: HomeAssistant) -> MockConfigEntry:
+    """Mock integration setup."""
+    config_entry = MockConfigEntry(
+        domain=BALBOA_DOMAIN,
+        data={
+            CONF_HOST: TEST_HOST,
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.balboa.BalboaSpaWifi",
+        new=BalboaMock,
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    return config_entry
+
+
+async def init_integration_mocked(hass: HomeAssistant) -> MockConfigEntry:
+    """Mock integration setup."""
+    config_entry = MockConfigEntry(
+        domain=BALBOA_DOMAIN,
+        data={
+            CONF_HOST: TEST_HOST,
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.connect",
+        new=BalboaMock.connect,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.listen_until_configured",
+        new=BalboaMock.listen_until_configured,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.listen",
+        new=BalboaMock.listen,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.check_connection_status",
+        new=BalboaMock.check_connection_status,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.send_panel_req",
+        new=BalboaMock.send_panel_req,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.send_mod_ident_req",
+        new=BalboaMock.send_mod_ident_req,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.spa_configured",
+        new=BalboaMock.spa_configured,
+    ), patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.get_model_name",
+        new=BalboaMock.get_model_name,
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    return config_entry
+
+
+class BalboaMock:
```

Comment:
I'd try to patch and autospec the library instead. Maintaining this large mock is fragile.